consensus/istanbul/backend/backend.go

```diff
@@ -497,9 +492,15 @@ func (sb *Backend) Commit(proposal istanbul.Proposal, aggregatedSeal types.Istan
 		return nil
 	}
 
-	if sb.broadcaster != nil {
-		sb.broadcaster.Enqueue(fetcherID, block)
+	// Note that result should NOT be nil, one edge case is when the validator re-start in the middle of a sequence.
+	// When happening, we re-verify the proposal and write it.
+	if result == nil {
+		sb.logger.Error("state process cache miss, verifying the block again", "number", block.Number(), "hash", block.Hash())
```

Comment:
several things here:

1. if `result != nil` it should still call `onNewConsensusBlock`
2. The log message implies that `backend` knows about the existence of a `cache`. That's a way of semantic coupling. It shouldn't be aware of any cache.
3. If `result ==nil` and `sb.Verify()` fails, we should return and error. As a rule of thumb, we should never silently ignore errors.

Reply:
The comment here is confusing to me. If the result should not be nil, then we shouldn't allow it to be nil, we should panic. If however it is a valid case that the result can be nil, then the comment needs to be changed, it could simply explain the case where the result is nil.